internal/resources/metal/connection/models.go

```diff
@@ -122,51 +125,67 @@ func (m *ResourceModel) parse(ctx context.Context, conn *packngo.Connection) dia
 	return diags
 }
 
+func getAbstractVirtualCircuit(vc *metalv1.VirtualCircuit) abstractVirtualCircuit {
```

Comment:
A similar method exists in the generated SDK: `GetActualInstance`
https://github.com/equinix/equinix-sdk-go/blob/main/services/metalv1/model_virtual_circuit.go#L96

Reply:
I took a similar approach here: https://github.com/equinix/docker-machine-driver-metal/pull/81/files#diff-9cd07bb5297426eece5e3a7661c840fa020615eb84ef7f2db03ccca07bf62f40R343-R363

Reply:
using GetAcutualInstance here now but still using that interface so I don't need to switch on a concrete type to get some fields shared between both